### PR TITLE
Components: Move `HappinessEngineersTray` to client, remove data-stores dep

### PR DESCRIPTION
**Pull request by tyxla**
## Proposed Changes

Originally started by @jsnajdr in https://github.com/Automattic/wp-calypso/pull/86536, this PR moves the `HappinessEngineersTray` component to Calypso's client as it's only used in the client since https://github.com/Automattic/wp-calypso/pull/63459. We also remove the `@automattic/data-stores` dependency from the components package and a non-working component example from devdocs. This is in preparation for releasing a new version of `@automattic/components`.

## Testing Instructions

* Verify all checks are green
* Verify devdocs UI components page loads and works well. 
* Verify the rest of the usages of `HappinessEngineersTray` work alright:
  *   `/hosting-config/:site`
  * `/home/:site` for a freshly purchased Creator plan.

### CODE REVIEW

File: client/devdocs/design/index.jsx [hunk 1]
```diff
@@ -237,7 +233,6 @@ export default class DesignAssets extends Component {
 					<LineChart readmeFilePath="line-chart" />
 					<LinkCard readmeFilePath="link-card" />
 					<ListEnd readmeFilePath="list-end" />
-					<ListTile readmeFilePath="/packages/components/src/list-tile" />
```
Comment (tyxla) [inline client/devdocs/design/index.jsx:240]: This was not working anyway, so removing.

File: client/me/help/help-happiness-engineers/index.jsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { HappinessEngineersTray } from '@automattic/components';
 import { localize } from 'i18n-calypso';
 import FormSectionHeading from 'calypso/components/forms/form-section-heading';
+import { HappinessEngineersTray } from 'calypso/components/happiness-engineers-tray';
```
Comment (tyxla) [inline client/me/help/help-happiness-engineers/index.jsx:4]: Actually, this component is also unused and can possibly be removed.

**Review by jsnajdr — APPROVED** [review]
:+1:

**Comment by tyxla** [comment]
Also removed the `wpcom-proxy-request` dependency as suggested by @jsnajdr via Slack.

**Comment by tyxla** [comment]
Verified that the moved component works well in the hosting config page:

happiness

File: client/me/help/help-happiness-engineers/index.jsx (hunk 2 above)
Comment (tyxla) [reply client/me/help/help-happiness-engineers/index.jsx:4]: Removed in https://github.com/Automattic/wp-calypso/pull/86544

**Pull request merged**